### ISSUE TITLE
Multiple identical data sources

### ISSUE THREAD

**Issue by djdeath** [google/perfetto#390]
I have a system with multiple GPUs (multiple of the same vendor).
I would like to create a data source for each of the GPU to report metrics, but it seems at the moment I have to create a data source class for each one.

I've tried to instantiate the data source class multiple times for each GPU, each with a different name, but it seems that the second instances are not visible to the perfetto system.

Is there a way to achieve this without creating multiple classes?

**Comment by LalitMaganti** [comment]
Can you provide more concrete examples of what your classes/configs look like? It's a bit hard to parse exactly what issue you are facing atm.

@ddiproietto FYI.

**Comment by djdeath** [comment]
Sure, I have this `GpuDataSource` class : https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/tool/pps/pps_datasource.h#L31

And what I would like to do it to register it multiple times like this : https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/tool/pps/pps_datasource.cc#L144

So far if I register more than once, even with different source name, the second one is never started, even if its name is in the perfetto config file.

**Comment by sashwinbalaji** [comment]
Sorry about the long delay but yeah this is a limitation of the C++ sdk. I have raised https://github.com/google/perfetto/pull/6600 to add better log and also documentation on ways this can be resolved.
